src/coreclr/interpreter/ee_il_dll.cpp

```diff
@@ -0,0 +1,142 @@
+#include <stdio.h>
+
+#ifdef _MSC_VER
+#define DLLEXPORT
+#else
+#define DLLEXPORT __attribute__ ((visibility ("default")))
+#endif // _MSC_VER
+
+#include <vector>
+
+// FIXME We will probably end up not needing this table.
+// If deemed useful, use some hashtable implementation instead.
+std::vector<std::pair<CORINFO_METHOD_HANDLE,InterpMethod*>> g_interpCodeHash;
+
+static InterpMethod* InterpGetInterpMethod(CORINFO_METHOD_HANDLE methodHnd)
+{
+    // FIXME lock for multiple thread access
+    for (int i = 0; i < g_interpCodeHash.size(); i++)
+    {
+        if (g_interpCodeHash[i].first == methodHnd)
+        {
+            return g_interpCodeHash[i].second;
+        }
+    }
+
+    InterpMethod* pMethod = new InterpMethod();
+    pMethod->methodHnd = methodHnd;
+
+    g_interpCodeHash.push_back({methodHnd, pMethod});
+    return pMethod;
+}
+
+/*****************************************************************************/
+ICorJitHost* g_interpHost        = nullptr;
+bool         g_interpInitialized = false;
+/*****************************************************************************/
+extern "C" DLLEXPORT void jitStartup(ICorJitHost* jitHost)
+{
+    if (g_interpInitialized)
+    {
+        return;
+    }
+    g_interpHost = jitHost;
+    // TODO Interp intialization
+    g_interpInitialized = true;
+}
+/*****************************************************************************/
+static CILInterp g_CILInterp;
+extern "C" DLLEXPORT ICorJitCompiler* getJit()
+{
+    if (!g_interpInitialized)
+    {
+        return nullptr;
+    }
+    return &g_CILInterp;
+}
+
+static InterpManager g_Manager;
+extern "C" DLLEXPORT ICorInterpreter* getInterpreter()
+{
+    if (!g_interpInitialized)
+    {
+        return nullptr;
+    }
+    return &g_Manager;
+}
+
+//****************************************************************************
+CorJitResult CILInterp::compileMethod(ICorJitInfo*         compHnd,
+                                   CORINFO_METHOD_INFO* methodInfo,
+                                   unsigned             flags,
+                                   uint8_t**            entryAddress,
+                                   uint32_t*            nativeSizeOfCode)
+{
+
+    const char *methodName = compHnd->getMethodNameFromMetadata(methodInfo->ftn, nullptr, nullptr, nullptr, 0);
+
+    // TODO: replace this by something like the JIT does to support multiple methods being specified and we don't
+    // keep fetching it on each call to compileMethod
+    const char *methodToInterpret = g_interpHost->getStringConfigValue("AltJit");
+    bool doInterpret = (methodName != NULL && strcmp(methodName, methodToInterpret) == 0);
+    g_interpHost->freeStringConfigValue(methodToInterpret);
+
+    if (!doInterpret)
+    {
+        return CORJIT_SKIPPED;
+    }
+
+    InterpMethod *pMethod = InterpGetInterpMethod(methodInfo->ftn);
+    pMethod->compiled = true;
+ 
+    compHnd->setMethodAttribs(methodInfo->ftn, CORINFO_FLG_INTERPRETER);
```

Comment:
Is this a race in the future if we set the INTERPRETER flag before we're done initializing all the interpreter state and metadata for the method? Or do we actually *want* to set it early so that once threading is wired up, other threads will see the flag and wait for us to finish compiling?

Reply:
This flag should not be set by the interpreter JIT in the first place. The synchronization for multiple threads trying to JIT a method at the same and other global state management should be done on the VM side around the call to the interpreter JIT, like it is done for regular JIT.

Reply:
Added a FIXME for now. Seems to me that this is something best refactored once we stop relying on the AltJit for kicking interpreter compilation.